server/src/main/java/com/distkv/server/Service.java

```diff
@@ -0,0 +1,48 @@
+package com.distkv.server;
+
+import java.io.Closeable;
+
+/**
+ * Service LifeCycle.
+ *
+ */
+public interface Service extends Closeable {
+
+  enum ServiceStatus {
+    UNINITED(0, "UNINITED"),
+    INITED(1, "INITED"),
+    RUNNING(2, "RUNNING"),
+    STOPPED(3, "STOPPED");
+
+    public final String stateName;
+    public final int value;
+
+    ServiceStatus(int value, String stateName) {
+      this.stateName = stateName;
+      this.value = value;
+    }
+  }
+
+  /**
+   * Get the service name.
+   * @return service name
+   */
+  String getName();
+
+  ServiceStatus getStatus();
+
+  /**
+   * apply the configuration and init service
+   */
+  void config();
```

Comment:
What does this method do?

Reply:
I think we should refactor configure class. every configure item should have blew field
+ configure key
+ default value
+ type (default String)
+ description

there is a configuration class read all configuration item from config file. the service class will call config() method to apply this service related config item.

Reply:
Why not invoke `setXxxx()` for config?

Reply:
config() represent the status of service lifecycle.  setXxx is a implement do configure. you can invoke setXxxx at config() method to acheive the goal of configuration.